nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -100,6 +100,20 @@ let
     eval -- "\$@"
   '';
 
+  nginxCommonHeaders = ''
+    add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
+  '' + lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+    add_header Alt-Svc 'h3=":443"; ma=86400';
+  '';
+
+  nginxProxyHeaders = ''
```

Comment:
All of these (except for the last one, `proxy_set_header Proxy "";`) seem to be covered by `services.nginx.recommendedProxySettings`, would setting that be sufficient?

Reply:
In `recommendedProxyConfig` there are unused parameters:
```
    proxy_set_header        X-Forwarded-Host $host;
    proxy_set_header        X-Forwarded-Server $host;
```
and there is no required parameter:
```
   proxy_set_header Proxy "";
```

Reply:
I would rather add the two unused header here, than maintaing a subset of those headers here.

Reply:
It will be difficult to predict the effect of this parameter in `recommendedProxyConfig` in configurations where it is used.

Reply:
It shouldn't. The extra headers should just be unused.

Reply:
If someone is using a `Proxy` header, it will reset the value.

Reply:
I don't think we generally consider double proxies. I would just go with recommendedProxyConfig and write a changelog entry that if you are double proxying, you need to change your config.